src/pages/result.tsx

```diff
@@ -1,53 +1,99 @@
 import { type PropsWithChildren } from "@kitajs/html";
 import Elysia from "elysia";
 import { type Session } from "lucia";
+import { HeaderHtml } from "../components/header";
 import { HxButton } from "../components/HxButton";
 import { LayoutHtml } from "../components/Layout";
 import { MatchDescription } from "../components/MatchDescription";
 import { NavbarHtml } from "../components/Navbar";
 import { ctx } from "../context";
-import { getMatches } from "../db/queries/matchQueries";
+import { getMatch, getMatches } from "../db/queries/matchQueries";
 import { getSeason } from "../db/queries/seasonQueries";
-import { getMatchRatingDiff, getRatingSystem } from "../lib/ratings/rating";
+import {
+  getMatchRatingDiff,
+  Match,
+  Rating,
+  RatingSystem,
+} from "../lib/ratings/rating";
 import { isDefined } from "../lib/utils";
+import { SeasonPicker } from "./admin/components/SeasonPicker";
 
 export const matchResult = new Elysia({
   prefix: "/result",
 })
   .use(ctx)
-  .get("/:seasonId/:matchId", ({ html, params, session }) => {
-    return html(
-      page(
-        session,
-        parseInt(params.seasonId, 10),
-        parseInt(params.matchId, 10),
-      ),
-    );
+  .get("/:matchId", ({ html, params, session, query, ratingSystem }) => {
+    // Cannot use "season" from context/middleware as we don't want the "current" season
+    const seasonId = query.season ? parseInt(query.season, 10) : undefined;
+    const matchId = parseInt(params.matchId, 10);
+    return html(page(session, matchId, seasonId, ratingSystem));
   });
 
 async function page(
   session: Session | null,
-  seasonId: number,
   matchId: number,
+  seasonId: number | undefined,
+  ratingSystem: RatingSystem<Rating>,
 ) {
+  const match = await getMatch(matchId, !!session?.user);
+  if (!match) {
+    return <LayoutHtml>Match does not exist</LayoutHtml>;
+  }
+
+  return (
+    <LayoutHtml>
+      <NavbarHtml session={session} activePage="result" />
+      <div class="flex flex-row justify-between">
+        <HeaderHtml className="px-0" title="Match result" />
+        <SeasonPicker
+          basePath={`/result/${matchId}`}
+          season={{ id: match.seasonId }}
```

Comment:
```suggestion
          season={{ id: seasonId ?? match.seasonId }}
```
Right now you choose something in the season picker and then it reverts back to whatever season the match has.